src/libAtomVM/nifs.c

```diff
@@ -2429,3 +2435,23 @@ static term nif_atomvm_read_priv(Context *ctx, int argc, term argv[])
         return UNDEFINED_ATOM;
     }
 }
+
+static term nif_console_print(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    
+    term t = argv[0];
+    VALIDATE_VALUE(t, term_is_string);
+    
+    int ok;
+    char *str = interop_term_to_string(t, &ok);
```

Comment:
what about using interop_write_iolist ?

Reply:
also as a further optimization we might check if given argument is a binary, in that case we might just use binary data without malloc & memcpy.

Reply:
Please have a look -- tried both.